ehrql/backends/emis.py

```diff
@@ -25,6 +25,7 @@ class EMISBackend(SQLBackend):
     display_name = "EMIS"
     query_engine_class = TrinoQueryEngine
     patient_join_column = "registration_id"
+    patient_join_column_type = str
```

Comment:
You're absolutely right that we need this and I'd completely forgotten about it; but it's going to be more complicated than this because we the code which works out what kind of Arrow schema to use for the results doesn't (and can't) know which particular backend is going to be in use when it runs. As this change stands, it will blow up when you try to write results to an Arrrow file (and possibly a CSV too, I can't remember how strict the CSV writer is).

There's an old ticket on this here which I'll update with my latest thinking:
https://github.com/opensafely-core/ehrql/issues/743

For now, I think we should pull this commit out (and I'll a reference to it from that ticket so we can re-use the work later). And then we can tackle it at some point as part of the "make EMIS actually work" project.